packages/apollo-gateway/src/index.ts

```diff
@@ -495,17 +502,25 @@ export class ApolloGateway implements GraphQLService {
       }
     }
 
+    const serviceMap: ServiceMap = Object.entries(this.serviceMap).reduce(
+      (serviceDataSources, [serviceName, { dataSource }]) => {
+        serviceDataSources[serviceName] = dataSource;
+        return serviceDataSources;
+      },
+      {} as ServiceMap,
```

Comment:
Let's use a `Object.create(null)` here so there's no prototype inheritance.  Otherwise, a `serviceName` of `hasOwnProperty` (or `constructor`, `__proto__`, etc.) might change, e.g., the otherwise expected dynamics of the object:

```suggestion
      Object.create(null) as ServiceMap,
```